meson.build

```diff
@@ -256,8 +256,6 @@ flags = ['-D_ISOC99_SOURCE', '-D_GNU_SOURCE',
 link_flags = []
 
 test_flags = ['-Werror=implicit-function-declaration',
-              '-Wno-error=deprecated-declarations',
-              '-Wno-error=unused-function',
```

Comment:
Should we really remove these from the default flags? I mean I don't personally compile with `werror` usually but perhaps someone does.

Reply:
If someone enabled `Werror`, they can enable `Wno-error` too. We don't enable `Werror` globally ever, so no need to clear that. And we don't have any unused-function warnings, and deprecated-declarations doesn't cover all deprecations anyway.